processor/src/host/mod.rs

```diff
@@ -28,16 +27,14 @@ pub use mast_forest_store::{MastForestStore, MemMastForestStore};
 /// 3. handling advice events (which internally mutates the advice provider), and
 /// 4. handling debug and trace events.
 pub trait Host {
-    type AdviceProvider: AdviceProvider;
-
     // REQUIRED METHODS
     // --------------------------------------------------------------------------------------------
 
     /// Returns a reference to the advice provider.
-    fn advice_provider(&self) -> &Self::AdviceProvider;
+    fn advice_provider(&self) -> &AdviceProvider;
 
     /// Returns a mutable reference to the advice provider.
-    fn advice_provider_mut(&mut self) -> &mut Self::AdviceProvider;
+    fn advice_provider_mut(&mut self) -> &mut AdviceProvider;
```

Comment:
As one of the next steps, we will move the advice provider out of the `Host` and into the `Process` struct - right?